bson/son.py

```diff
@@ -136,51 +143,51 @@ def update(self, other=None, **kwargs):
         if kwargs:
             self.update(kwargs)
 
-    def get(self, key, default=None):
+    def get(self, key: _Key, default: Optional[Union[_Value, _T]] = None) -> Union[_Value, _T, None]:  # type: ignore[override]
         try:
             return self[key]
         except KeyError:
             return default
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
         """Comparison to another SON is order-sensitive while comparison to a
         regular dictionary is order-insensitive.
         """
         if isinstance(other, SON):
             return len(self) == len(other) and list(self.items()) == \
                    list(other.items())
-        return self.to_dict() == other
+        return cast(bool, self.to_dict() == other)
 
-    def __ne__(self, other):
+    def __ne__(self, other: Any) -> bool:
         return not self == other
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self.__keys)
 
-    def to_dict(self):
+    def to_dict(self) -> Dict[_Key, _Value]:
         """Convert a SON document to a normal Python dictionary instance.
 
         This is trickier than just *dict(...)* because it needs to be
         recursive.
         """
 
-        def transform_value(value):
+        def transform_value(value: Union[List[_Value], Dict[_Key, _Value]]) -> Union[List[_Value], Dict[_Key, _Value]]:
             if isinstance(value, list):
-                return [transform_value(v) for v in value]
+                return [transform_value(v) for v in value]  # type: ignore
             elif isinstance(value, _Mapping):
                 return dict([
-                    (k, transform_value(v))
+                    (k, transform_value(v))  # type: ignore
                     for k, v in value.items()])
             else:
                 return value
 
-        return transform_value(dict(self))
+        return cast(Dict[_Key, _Value], transform_value(dict(self)))
 
-    def __deepcopy__(self, memo):
-        out = SON()
+    def __deepcopy__(self, memo: Any) -> "SON[Any, Any]":
```

Comment:
I think we can clean this up and revmoe the `cast` like this:
```
    def __deepcopy__(self, memo: MutableMapping[int, "SON[_Key, _Value]"]) -> "SON[_Key, _Value]":

```

Reply:
Updated